test/unit/index.js

```diff
@@ -255,5 +255,54 @@ describe('Builder', function () {
 					done();
 				});
 		});
+
+		it('should pass results to .then() instead of a callback', function (done) {
```

Comment:
We should also have a callback that checks the analyze callback is executed before `.then()`.